### ISSUE TITLE
[Question] Parameter adjustments on training male voice with mb_melgan conf 

### ISSUE THREAD

**Issue by windtoker** [kan-bayashi/ParallelWaveGAN#196]
First of all, let me say I'm really grateful for your work. Really using this repo well and achieved a lot from here. Great work!

So the problem is that while training voices with mb_melgan, I find it hard to train  with male voices rather than female voices in mb-melgan. It shows good quality in female voice while not in male voice.

I trained single male voice (15 hours, librivox meian) in vctk multi_band_melgan.v2.yaml. i converted the sampling rate from 22050Hz to 24KHz and the sampling bit from 32 to 16. In the synthesis step, I transformed the original audio to Mel by espnet STFT first and apply it to the trained vocoder as to avoid any contribution from acoustic model. The sound quality is a bit disppointing as i could hear some howling noises with checkboard pattern in it.

I used default vctk multi_band_melgan.v2.yaml and found out the parameter settings in vctk is as same as those of ljspeech which is female dataset except fft_size, hop_size, and win_length.
I wanna make some parameter adjustments that would fit into male voice dataset. 
So I am asking you for advice on this matter.
1) Do you think those discrepant parameters above would have affected the differing qualities regarding gender?
2) I feel like I have to somehow touch discriminator and stft loss settings. do you think this is the right way?

Thanks in advance.

**Comment by kan-bayashi** [comment]
Hi @windtoker.

Actually, I have no so much experience with male dataset since there is no good single-male speaker free corpus, but I will try to answer your question.

> In the synthesis step, I transformed the original audio to Mel by espnet STFT first and apply it to the trained vocoder as to avoid any contribution from acoustic model. 

Do you mean the synthesis with the natural features (analysis-synthesis)?

> I used default vctk multi_band_melgan.v2.yaml and found out the parameter settings in vctk is as same as those of ljspeech which is female dataset except fft_size, hop_size, and win_length.

Yes but `upsample_scales` is also different.

> Do you think those discrepant parameters above would have affected the differing qualities regarding gender?

If the f0 of the speaker is very low, it is better to tune `fmin` (default=80).
And in general, it is better to remove the silence as much as possible, especially in the beginning and the end of utterances. For example, in my VCTK recipe, I use force alignment results to remove the silence.

> I feel like I have to somehow touch discriminator and stft loss settings. 

`lambda_adv` will affect the performance. In my experience, small `lambda_adv` decreases performance.
And I saw some papers using more multi-resolution STFT loss with different window/shift settings (currently 3 types).

And another important setting is optimizer settings.
You can see some discussion in #143.

**Comment by windtoker** [comment]
Thanks for the comment.

Do you mean the synthesis with the natural features (analysis-synthesis)?
-> yes. I guess what you are saying is what I meant.

Considering the proposals with linked issue above, guess I will lower the fmin range to 0 and increase lambda_adv a bit. also gonna fine-tune PQMF filter configuration look for some references regarding multiple stft loss.

thanks again!
